lib/LedgerSMB/Scripts/setup.pm

```diff
@@ -896,41 +892,45 @@ sub fix_tests{
 
 
     my $table = $check->table;
-    my $where = $check->id_where;
     my @edits = @{$check->columns};
     # If we are inserting and id is displayed, we want to insert
     # at this exact location
+    my $id_columns = join('|',@{$check->id_columns});
     my $id_displayed = $check->{insert}
-                and grep( /^$check->{id_column}$/, @{$check->display_cols} );
+                and grep( /^($id_columns)$/, @{$check->display_cols} );
 
     my $query;
     if ($check->{insert}) {
         my @_edits = @edits;
-        unshift @_edits, $check->{id_column} if $id_displayed;
+        unshift @_edits, @{$check->id_columns} if $id_displayed;
```

Comment:
I'm unfamiliar with the UI, but why doesn't the UI provide us with these values? Isn't the key to the situation that the row is *new*, so the p-key isn't available in advance?